internal/engine/compiler/cpuid_amd64.go

```diff
@@ -0,0 +1,43 @@
+package compiler
```

Comment:
can we move this to under platform/**? Then I could reuse it later on the new backend